docs/DeveloperGuide.md

```diff
@@ -1,38 +1,150 @@
 # Developer Guide
```

Comment:
Consider adding class diagrams accompanied with object diagrams as a method to elaborate on certain aspects and functionalities of the application